modules/auxiliary/gather/jenkins_cred_recovery.rb

```diff
@@ -0,0 +1,276 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'json'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Jenkins Domain Credential Recovery',
+      'Description'    => %q{
+        This module will collect Jenkins domain credentials, and uses
+        the script console to decrypt each password if anonymous permission
+        is allowed.
+
+        It has been tested against Jenkins version 1.590, 1.633, and 1.638.
+      },
+      'Author'         =>
+        [
+          'Th3R3p0', # Vuln Discovery, PoC
+          'sinn3r'   # Metasploit
+        ],
+      'References'     =>
+        [
+          [ 'EDB', '38664' ],
+          [ 'URL', 'http://www.th3r3p0.com/vulns/jenkins/jenkinsVuln.html' ]
+        ],
+      'DefaultOptions' =>
+        {
+          'RPORT' => 8080
+        },
+      'License'        => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI',     [true, 'The base path for Jenkins', '/']),
+        OptString.new('JENKINSDOMAIN', [true, 'The domain where we want to extract credentials from', '_'])
+      ], self.class)
+  end
+
+
+  # Returns the Jenkins version.
+  #
+  # @return [String] Jenkins version.
+  # @return [NilClass] No Jenkins version found.
+  def get_jenkins_version
+    uri = normalize_uri(target_uri.path)
+    res = send_request_cgi({ 'uri' => uri })
+    html = res.get_html_document
```

Comment:
heheh, I was expecting someone to point this point. But nah, #get_html_document is nil-safe:

```
msf sinn3r $ irb
2.1.7 :001 > require 'nokogiri'
 => true 
2.1.7 :002 > Nokogiri::HTML(nil)
 => #<Nokogiri::HTML::Document:0x3fcb0984b188 name="document" children=[#<Nokogiri::XML::DTD:0x3fcb0984adc8 name="html">]> 
```

So you won't error out or backtrace due to a nil :-)